src/engine/Server.cpp

```diff
@@ -520,13 +521,28 @@ Awaitable<std::function<void()>> Server::cancelAfterDeadline(
       pointer->abort(ad_utility::AbortionState::TIMEOUT);
     }
   };
-  net::co_spawn(executor, cancelAfterTimeout(std::move(abortionHandle), timer),
+  net::co_spawn(strand, cancelAfterTimeout(std::move(abortionHandle), timer),
                 net::detached);
-  co_return [executor, timer = std::move(timer)]() mutable {
-    net::post(executor, [timer = std::move(timer)]() { timer->cancel(); });
+  return [strand, timer = std::move(timer)]() mutable {
+    net::post(strand, [timer = std::move(timer)]() { timer->cancel(); });
   };
 }
 
+// ____________________________________________________________________________
+std::function<void()> Server::setupAbortionHandle(
```

Comment:
I think the return type here can also be `auto`. 
And actually, to better specify that the function returns a lambda with certain arguments, we can constrain the return type:

```suggestion
std::invocable auto Server::setupAbortionHandle(
```

(same for the function above).
Be the first one to use concepts in this way in QLever:)

Reply:
That does not work because 2 different lambdas will be returned, so when using auto, the types are conflicting